vue.config.js

```diff
@@ -31,5 +32,16 @@ module.exports = vueUtils({
         toType: 'dir',
         ignore: ['.DS_Store'],
       }]]);
+
+    // limit highlight.js to only the necessary languages
+    const builtinLanguages = 'bash|c|s?css|cpp|diff|http|java|llvm|perl|php|python|ruby|xml|javascript|json|markdown|objectivec|shell|swift';
+    const envLanguages = (process.env.VUE_APP_HLJS_LANGUAGES ?? '').split(',').join('|');
+    config
+      .plugin('languages')
+      .use(webpack.ContextReplacementPlugin, [
+        /highlight\.js\/lib\/languages$/,
+        new RegExp(`\/(${[builtinLanguages, envLanguages].join('|')})$`),
+        /\.js$/,
```

Comment:
i think the `.js` part should be inside the regex, no? Otherwise that is the third parameter to `ContextReplacementPlugin`, which seems to be a boolean - https://webpack.js.org/plugins/context-replacement-plugin/#usage

Example of what you did:

https://github.com/date-fns/date-fns/blob/main/docs/webpack.md